packages/flutter_tools/lib/src/flutter_plugins.dart

```diff
@@ -1433,7 +1433,7 @@ bool _hasPluginInlineImpl(
 /// Determine if the plugin provides an inline Dart implementation.
 bool _hasPluginInlineDartImpl(Plugin plugin, String platformKey) {
   return plugin.pluginDartClassPlatforms[platformKey] != null &&
-      plugin.pluginDartClassPlatforms[platformKey] != 'none';
+      plugin.pluginDartClassPlatforms[platformKey]!.dartClass != 'none';
```

Comment:
The lookup should be extracted to a local variable to avoid the dangerous force-unwrap.

Reply:
Done.